controlplane/kubeadm/controllers/controller.go

```diff
@@ -212,6 +212,27 @@ func (r *KubeadmControlPlaneReconciler) Reconcile(req ctrl.Request) (res ctrl.Re
 	return r.reconcile(ctx, cluster, kcp)
 }
 
+// setSingleMachineConditions updates the machine's conditions according to health tracker.
+func setSingleMachineConditions(machine *clusterv1.Machine, controlPlane *internal.ControlPlane) {
+	for condType, condition := range controlPlane.MachineConditions[machine.Name] {
+		doesConditionExist := false
+		for _, mCondition := range machine.Status.Conditions {
+			// If the condition already exists, change the condition.
+			if mCondition.Type == condType {
+				conditions.Set(machine, condition)
+				doesConditionExist = true
+			}
+		}
+		if !doesConditionExist {
+			if machine.Status.Conditions == nil {
+				machine.Status.Conditions = clusterv1.Conditions{}
+			}
+			conditions.Set(machine, condition)
+		}
+
+	}
+}
```

Comment:
We should be able to remove this and just set the conditions on the Machine itself when required